### PR TITLE
add barber smooth function

### PR DESCRIPTION
**Pull request by aucarter**

### CODE REVIEW

File: tests/testthat/test_barber_smooth.R [hunk 1]
```diff
@@ -0,0 +1,12 @@
+test_that("The function runs", {
+  vec <- c(1, 7, 5, 6, 19, 10)
+  smoothed <- barber_smooth(vec, n_neighbors = 2, times = 1)
+
+  expect_equal(length(smoothed), 6)
+  expect_equal(class(smoothed), "matrix")
```
Comment (miker985) [inline tests/testthat/test_barber_smooth.R:6]: @aucarter do you know if this should be converting the result back to a vector?

File: tests/testthat/test_barber_smooth.R [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  expect_equal(
+    smoothed,
+    matrix(data = c(2.79, 3.81, 5.25, 8.32, 8.69, 10.93)),
```
Comment (miker985) [inline tests/testthat/test_barber_smooth.R:9]: I literally just pasted the actual results to create this test data. I assume this is reasonable. Please give it a quick glance

File: tests/testthat/test_barber_smooth.R (hunk 1 above)
Comment (aucarter) [reply tests/testthat/test_barber_smooth.R:6]: it probably should be a vector, but is not utilized in any ways where that matters. i am going to leave it as is for now.

File: tests/testthat/test_barber_smooth.R (hunk 2 above)
Comment (aucarter) [reply tests/testthat/test_barber_smooth.R:9]: yes. this is very reasonable.

**Review by miker985 — APPROVED** [review]

**Pull request merged**